### DISCUSSION TITLE
Does Apollo Server take advantage of multicore containers?

### DISCUSSION THREAD

**Discussion by ayrtonvwf** [apollographql/apollo-server#5134] [Q&A]
I deploy Apollo Federation with Google Cloud Run.
Cloud Run gives me the option to set the "Number of vCPUs allocated to each container instance", where I can choose from 1, 2 or 4 CPUs.
Does Apollo Server take advantage of a multicore deployment?
I know that Node.JS is single-threaded but it does have Worker theads that would make it possible. Does Apollo Server do that?

**Comment by glasser** [comment]
Apollo Server doesn't have anything in particular inside it that takes advantage of any extra concurrency models like worker threads.
